dbt/scripts/run_iasworld_data_tests.py

```diff
@@ -733,35 +720,73 @@ def main() -> None:
             "including null townships (which typically indicate invalid PINs)."
         ),
     )
+    parser.add_argument(
+        "--use-cached",
+        action=argparse.BooleanOptionalAction,
+        required=False,
+        help=(
+            "Toggle using cached results from the most recent run. Useful when debugging "
+            "transformation steps. Defaults to False."
+        ),
+    )
 
     args = parser.parse_args()
 
-    run_results_filepath = args.run_results
-    manifest_filepath = args.manifest
     output_dir = args.output_dir
     townships = args.township if args.township else tuple()
+    use_cached = args.use_cached
+
+    run_results_filepath = os.path.join("target", "run_results.json")
+    manifest_filepath = os.path.join("target", "manifest.json")
 
     date_today = datetime.datetime.today().strftime("%Y-%m-%d")
     if output_dir is None:
-        output_dir = f"qc_test_results_{date_today}"
-
-    test_cache_path = get_test_cache_path(
-        run_results_filepath, manifest_filepath, townships
-    )
+        output_dir = f"iasworld_test_results_{date_today}"
 
-    if os.path.isfile(test_cache_path):
-        print(f"Loading test results from cache at {test_cache_path}")
-        test_categories = get_test_categories_from_file(test_cache_path)
-    else:
-        print(
-            f"Test cache not found at {test_cache_path}, loading test results "
-            "from Athena"
+    if use_cached:
+        test_cache_path = get_test_cache_path(
+            run_results_filepath, manifest_filepath, townships
         )
+        if os.path.isfile(test_cache_path):
+            print(f"Loading test results from cache at {test_cache_path}")
+            test_categories = get_test_categories_from_file(test_cache_path)
+        else:
+            raise ValueError(
+                f"Test cache not found at {test_cache_path}, try rerunning "
+                "without --use-cached"
+            )
+    else:
+        print("Running tests")
+        dbt_run_args = [
+            "test",
+            "--selector",
+            "data_test_iasworld",
+            "--store-failures",
+        ]
+        print(f"> dbt {' '.join(dbt_run_args)}")
+        dbt_test_result = DBT.invoke(dbt_run_args)
+
+        if dbt_test_result.exception is not None:
+            raise dbt_test_result.exception
+
+        if any(
+            result.status == TestStatus.Error
+            for result in getattr(dbt_test_result.result, "results", [])
+        ):
+            # No need to report the exception, since the dbt process
+            # will have printed it already
+            raise ValueError("Quitting due to error in dbt test run")
```

Comment:
This block replaces the old `Run tests` step in the `test-dbt-models` workflow, including the error handling.

Reply:
Yeah this is way way better. <3 Python